test/unit/chatAgentRequestHandler.test.ts

```diff
@@ -0,0 +1,392 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from "vscode";
+import * as TypeMoq from "typemoq";
+import { expect } from "chai";
+import { createSqlAgentRequestHandler } from "../../src/chat/chatAgentRequestHandler";
+import { CopilotService } from "../../src/services/copilotService";
+import VscodeWrapper from "../../src/controllers/vscodeWrapper";
+import * as Utils from "../../src/models/utils";
+import * as sinon from "sinon";
+import * as telemetry from "../../src/telemetry/telemetry";
+import { GetNextMessageResponse, MessageType } from "../../src/models/contracts/copilot";
+import { ActivityObject, ActivityStatus } from "../../src/sharedInterfaces/telemetry";
+
+suite("Chat Agent Request Handler Tests", () => {
+    let mockCopilotService: TypeMoq.IMock<CopilotService>;
+    let mockVscodeWrapper: TypeMoq.IMock<VscodeWrapper>;
+    let mockContext: TypeMoq.IMock<vscode.ExtensionContext>;
+    let mockLmChat: TypeMoq.IMock<vscode.LanguageModelChat>;
+    let mockChatStream: TypeMoq.IMock<vscode.ChatResponseStream>;
+    let mockChatRequest: TypeMoq.IMock<vscode.ChatRequest>;
+    let mockChatContext: TypeMoq.IMock<vscode.ChatContext>;
+    let mockToken: TypeMoq.IMock<vscode.CancellationToken>;
+    let mockTextDocument: TypeMoq.IMock<vscode.TextDocument>;
+    let mockConfiguration: TypeMoq.IMock<vscode.WorkspaceConfiguration>;
+    let mockLanguageModelChatResponse: TypeMoq.IMock<vscode.LanguageModelChatResponse>;
+    let mockActivityObject: TypeMoq.IMock<ActivityObject>;
+    let generateGuidStub: sinon.SinonStub;
+    let selectChatModelsStub: sinon.SinonStub;
+    let startActivityStub: sinon.SinonStub;
+    let sendActionEventStub: sinon.SinonStub;
+    let openTextDocumentStub: sinon.SinonStub;
+
+    // Sample data for tests
+    const sampleConnectionUri = "file:///path/to/sample.sql";
+    const sampleConversationUri = "conversationUri1";
+    const samplePrompt = "Tell me about my database schema";
+    const sampleCorrelationId = "12345678-1234-1234-1234-123456789012";
+    const sampleReplyText = "Here is information about your database schema";
+
+    setup(() => {
+        // Create the mock activity object for startActivity to return
```

Comment:
if you make these stubs in a sandbox, the teardown reset becomes much cleaner/simpler.  See `addFirewallRuleWebviewController.test.ts`